src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -788,6 +788,24 @@ private bool IsValidExtensionMethodSignature()
                 this.ParameterRefKinds.IsDefault && // No 'ref' or 'out'
                 !this.IsParams();
 
+        private bool IsValidUserDefinedOperatorIs()
+        {
+            foreach (var parameter in this.Parameters)
+            {
+                if (parameter.RefKind != ((parameter.Ordinal == 0) ? RefKind.None : RefKind.Out))
+                {
+                    return false;
+                }
+            }
+
+            return
+                (this.ReturnsVoid || this.ReturnType.SpecialType != SpecialType.System_Boolean) &&
+                !this.IsGenericMethod &&
+                !this.IsVararg &&
+                this.ParameterCount > 0 &&
```

Comment:
I do not intend the requirements to be the same, as they serve different purposes. Please open a github issue for the design change you'd like in the compilers and that can be considered separately. It is outside the scope of this feature.